pkg/resource/queue/sdk.go

```diff
@@ -292,6 +292,15 @@ func (rm *resourceManager) sdkUpdate(
 	if err != nil {
 		return nil, err
 	}
+	// note(Julian-Chu): SetAttributes API without any attributes
+	// will return MalformedInput(message: End of list found where not expected) error.
+	// if there are no attributes in the input,
+	// We need to set minimal one default value, or use customUpdate to skip api call.
+	if len(input.Attributes) == 0 {
+		input.Attributes = map[string]*string{
+			"DelaySeconds": latest.ko.Spec.DelaySeconds,
+		}
+	}
```

Comment:
shouldn't we fix the delta function instead? Also i don't understand how we get to this block when we're having:
```
	if delta.DifferentAt("Spec.Tags") {
		if err := rm.syncTags(ctx, desired, latest); err != nil {
			return nil, err
		}
	}
	if !delta.DifferentExcept("Spec.Tags") {
		return desired, nil
	}
```

Reply:
!! you're right, we should fix delta.

I quickly check the delta:
![image](https://github.com/aws-controllers-k8s/sqs-controller/assets/7224544/4c2dde88-02c9-4049-9d74-be657cdb6e27)
looks some default values returned from API.

I will find some time next week to fix it.